src/main/java/net/snowflake/ingest/streaming/internal/AbstractRowBuffer.java

```diff
@@ -427,7 +432,7 @@ Set<String> verifyInputColumns(
     List<String> missingCols = new ArrayList<>();
     for (String columnName : this.nonNullableFieldNames) {
       if (!inputColNamesMap.containsKey(columnName)) {
-        missingCols.add(statsMap.get(columnName).getColumnDisplayName());
+        missingCols.add(fieldIndex.get(columnName).columnMetadata.getName());
```

Comment:
actually, is there a need to even make this change?

Reply:
If a root column is a structured data type, `statsMap.get(columnName)` will trow NPE as we only stores leaf column stats. Use `fieldIndex` instead for logging to avoid this.

Reply:
please add this as a code comment.